jetty-ee10/jetty-ee10-osgi/test-jetty-ee10-osgi/src/test/java/org/eclipse/jetty/ee10/osgi/test/TestOSGiUtil.java

```diff
@@ -66,54 +69,67 @@ public void stop(BundleContext context) throws Exception
-            xmlConfigs.append(";");
-            xmlConfigs.append(new File(etc, "jetty-alpn.xml").toURI());
-            xmlConfigs.append(";");
-            xmlConfigs.append(new File(etc, "jetty-https.xml").toURI());
-            xmlConfigs.append(";");
+            xmlFiles.add(etc.resolve("jetty-ssl.xml"));
+            xmlFiles.add(etc.resolve("jetty-ssl-context.xml"));
+            xmlFiles.add(etc.resolve("jetty-alpn.xml"));
+            xmlFiles.add(etc.resolve("jetty-https.xml"));
         }
-        xmlConfigs.append(new File(etc, jettySelectorFileName).toURI());
-        xmlConfigs.append(";");
-        xmlConfigs.append(new File(etc, "jetty-deploy.xml").toURI());
-        xmlConfigs.append(";");
-        xmlConfigs.append(new File(etc, "jetty-testrealm.xml").toURI());
+        xmlFiles.add(etc.resolve(jettySelectorFileName));
+        xmlFiles.add(etc.resolve("jetty-deploy.xml"));
+        xmlFiles.add(etc.resolve("jetty-testrealm.xml"));
+
+        // Sanity check that the referenced files actually exist.
+        xmlFiles.forEach(p -> assertTrue(p + " doesn't exists and/or isn't a file", Files.isRegularFile(p)));
+
+        // Convert list of XML to a list of URIs separated by ";" (not OS specific, but OSGI specific)
+        String xmlConfigLine = xmlFiles.stream()
+            .map(Path::toUri)
+            .map(URI::toASCIIString)
+            .collect(Collectors.joining(";"));
 
-        options.add(systemProperty(OSGiServerConstants.MANAGED_JETTY_XML_CONFIG_URLS).value(xmlConfigs.toString()));
+        options.add(systemProperty(OSGiServerConstants.MANAGED_JETTY_XML_CONFIG_URLS).value(xmlConfigLine));
         options.add(systemProperty("jetty.http.port").value("0"));
-        options.add(systemProperty("jetty.home").value(etc.getParentFile().getAbsolutePath()));
-        options.add(systemProperty("jetty.base").value(etc.getParentFile().getAbsolutePath()));
+        options.add(systemProperty("jetty.home").value(etc.getParent().toString()));
+        options.add(systemProperty("jetty.base").value(etc.getParent().toString()));
         return options;
     }
 
     public static List<Option>  configureJettyHomeAndPortViaBootBundle(String jettyConnectorListenerFileName)
     {
-        //use the default set of config files embedded in jetty boot jar
         List<Option> options = new ArrayList<>();
-        StringBuffer xmlConfigs = new StringBuffer();
-        xmlConfigs.append(JettyBootstrapActivator.DEFAULT_JETTY_ETC_FILES);
-        xmlConfigs.append(",");
-        //add in a couple of external files needed for testing
-        File etc = new File(FS.separators("src/test/config/etc"));
-        xmlConfigs.append(new File(etc, jettyConnectorListenerFileName).toURI());
-        xmlConfigs.append(",");
-        xmlConfigs.append(new File(etc, "jetty-testrealm.xml").toURI());
+        List<Path> xmlFiles = new ArrayList<>();
+        Path etc = MavenPaths.projectBase().resolve("src/test/config/etc");
```

Comment:
This is incorrect. What we are doing here is using the `etc` files embedded inside the jetty bundle jar, _not_ the ones in the `test` directory.

Reply:
The previous code did the same thing this code does.

``` java
        File etc = new File(FS.separators("src/test/config/etc"));
        xmlConfigs.append(new File(etc, jettyConnectorListenerFileName).toURI());
        xmlConfigs.append(",");
        xmlConfigs.append(new File(etc, "jetty-testrealm.xml").toURI());
```

Reply:
You might be thinking of line 120 ...

``` java
        xmlReferences.addAll(JettyBootstrapActivator.DEFAULT_JETTY_XML_FILES);
```

Reply:
I reorganized the methods so that you can follow them better.
But the end result is still the same.
XML references into bundle first, followed by file URI references, in the paxexam options.

This `xmlConfigLine` is identical now to how it was when this PR was opened.

``` java
options.add(systemProperty(OSGiServerConstants.MANAGED_JETTY_XML_CONFIG_URLS).value(xmlConfigLine));
```